### ISSUE TITLE
Labels.. funniest thing to make my day

### ISSUE THREAD

**Issue by oziee** [photoprism/photoprism#160]
More with my testing, I have found Labels to be so wrong I had a good 30mins of laughing looking through the list of items it has set for photos. 

They were so wrong that labels are not useful at all.

Can we turn labels off? I have seem other posts on TF models and looking forward to changes with this (one day people tag with recognition/identification)

**Comment by lastzero** [comment]
You mean turn off titles or the labels page? Labels in general can have a very low probability like 8%. Also they have different priorities. If you have a window and a cat on a photo, you are probably more interested in the cat.

**Comment by lastzero** [comment]
Maybe we should limit what the user sees or just show categories instead of  individual labels? For development, it's good to see raw data though.

**Comment by oziee** [comment]
here are some examples of things i can put in public. nothing in the photos fits with the labels :)

<img width="466" alt="Screen Shot 2019-12-15 at 4 46 35 pm" src="https://user-images.githubusercontent.com/1471841/70859158-aa209080-1f5a-11ea-81f5-e28b820588f1.png">
<img width="234" alt="Screen Shot 2019-12-15 at 4 46 06 pm" src="https://user-images.githubusercontent.com/1471841/70859159-aab92700-1f5a-11ea-96f3-73b4c57a6ad7.png">
<img width="232" alt="Screen Shot 2019-12-15 at 4 45 56 pm" src="https://user-images.githubusercontent.com/1471841/70859161-aab92700-1f5a-11ea-81e7-7b188d7e06ee.png">

**Comment by lastzero** [comment]
Well, those labels make sense in a certain way. But they shouldn't be used with high priority. Once you have a lot of photos, you would use the one with the highest probability as preview which hopefully makes it look less funny.

**Comment by lastzero** [comment]
Does this go in the right direction? You can use our [demo](https://demo.photoprism.org/) for testing.

**Comment by lastzero** [comment]
I'll close this. Let us know if you find more labels that need tweaking. It's a lot of work. Most people think you just need to use a TensorFlow model, but it's a lot of linguistics too. A TensorFlow classification is really just an ID and a probability.

**Comment by lastzero** [comment]
Spent 14 hours today refactoring labels... should be way faster now and less labels should be visible by default. You can't delete existing labels (as they are in the model), but you can favor the labels you find most useful. Demo build is running.

![Screenshot 2019-12-16 at 23 57 17](https://user-images.githubusercontent.com/301686/70950071-d114ba80-205f-11ea-9185-579f9402e67f.png)

**Comment by oziee** [comment]
nope sorry.. labels are still terrible and nothing like what is in the images

**Comment by lastzero** [comment]
Maybe you can be more specific... of course if you only have photos of cables and random objects, this does not work as expected. But animals for example work pretty well for us.

**Comment by oziee** [comment]
no elephants in any photos, but it thinks there is

<img width="1398" alt="Screen Shot 2020-01-07 at 10 31 29 pm" src="https://user-images.githubusercontent.com/1471841/71896159-ad481e80-319e-11ea-81ab-691eec1fc461.png">

**Comment by lastzero** [comment]
Did you really update, delete the db and reindex? I personally disabled or changed most of those labels.

**Comment by lastzero** [comment]
`make install` will not delete the `labels.yml` in your config directory as you could have local changes... so you need to do `clean-install` or update `labels.yml` manually.

**Comment by oziee** [comment]
sure did.. deleted all files and now even the thumbs cache now that i know where it is:)
all directories were removed.. and a full github download and make install ran

**Comment by lastzero** [comment]
Check `~/.config/photoprism/`

**Comment by lastzero** [comment]
It great you're using a local installation as Docker users would not run into this

**Comment by oziee** [comment]
ok deleted it for sure this time coz i have to run make install again

yup, only 11 labels this time round.. and most fit well :)

**Comment by lastzero** [comment]
Think we'll compile this in, most users will not want to maintain this manually... it's better to tell us the problem and get an update :)

**Comment by xaviergxf** [comment]
@lastzero Is this change done on the docker images as well? I downloaded yesterday from docker hub the images and i still see some bizarre labels on my images.

**Comment by lastzero** [comment]
Please report with examples :) We continuously improve labels when we find issues!

**Comment by danielo515** [comment]
I would love to be able to fine-tune the labels via the UI.
Also, is there a way I can assign certain labels to some of my pictures? I understand that I can't "train" the model this way because it is already compiled, but will help me organise.

**Comment by lastzero** [comment]
@danielo515 Fine-tuning TensorFlow labels is beyond the scope of our first release. It took us months with very intensive testing to get this right, most users wouldn't have the time and knowledge to fine-tune this with correct mapping, linguistic tweaks and appropriate probability thresholds, see https://docs.photoprism.org/developer-guide/metadata/classification/.

You may assign labels to pictures in our edit dialog. Batch editing is on our todo, see #271.

**Comment by danielo515** [comment]
@lastzero sorry for not being clear about what I meant by fine-tune.
I saw you have some labels/tags that refers to others, for example in case of ant, you just refer it to insect. In my case, I have lots of pictures of ants, and I would love to have a dedicated auto recognized tag. Those mappings is what I may want to "ajdust"
